generator/default.p9k

```diff
@@ -0,0 +1,606 @@
+  #    highest precedence.
+  local CONDITION
+  local SEGMENT_CONDITION="P9K_${STATEFUL_NAME}_CONDITION"
+  if defined "${SEGMENT_CONDITION}"; then
+    CONDITION="${(P)SEGMENT_CONDITION}"
+  elif [[ -n "${10}" ]]; then
+    CONDITION="${10}"
+  else
+    CONDITION='[[ -n "${8}" ]]'
+  fi
+  # Precompile condition.
+  eval "${CONDITION}" && CONDITION=true || CONDITION=false
+
+#  if ! ${CONDITION}; then
+#    continue
+#  fi
+  [[ $CONDITION == true ]] && "$3PromptSegment" "${NAME}" "${INDEX}" "${BACKGROUND}" "${FOREGROUND}" "${BOLD}" "${CONTENT}" "${VISUAL_IDENTIFIER}"
+}
+
+################################################################
+# Prompt Segment Definitions
+################################################################
+
+# The `CURRENT_BG` variable is used to remember what the last BG color used was
+# when building the left-hand prompt. Because the RPROMPT is created from
+# right-left but reads the opposite, this isn't necessary for the other side.
+CURRENT_BG='NONE'
+
+################################################################
+# @description
+#   The `custom` prompt provides a way for users to invoke commands and display
+#   the output in a segment.
+##
+# @args
+#   $1 string Left|Right
+#   $2 integer Segment index
+#   $3 boolean Whether the segment should be joined
+#   $4 string Custom segment name
+##
+prompt_custom() {
+  local command=P9K_CUSTOM_$4:u
+  local segment_content="$(eval ${(P)command})"
+
+  if [[ -n $segment_content ]]; then
+    serializeSegment "${4:u}" "" "$1" $2 $3 $DEFAULT_COLOR_INVERTED $DEFAULT_COLOR "$segment_content"
+  fi
+}
+
+################################################################
+# Prompt processing and drawing
+################################################################
+
+###############################################################
+# @description
+#   This function loops through the left prompt elements and calls
+#   the related segment functions.
+##
+# @noargs
+##
+buildLeftPrompt() {
+  local index=1
+  local element joined
+  for element in "${P9K_LEFT_PROMPT_ELEMENTS[@]}"; do
+    # Check if segment should be joined
+    [[ "${element[-7,-1]}" == '_joined' ]] && joined=true || joined=false
+
+    # Remove joined information in direct calls
+    element="${element%_joined}"
+
+    # Check if it is a custom command, otherwise interpet it as
+    # a prompt.
+    if [[ $element[0,7] =~ "custom_" ]]; then
+      if [[ -z "${loadedSegments[(r)$element]}" ]]; then
+        if [[ -n "$P9K_${(U)element}" ]]; then # check if custom segment is a definition
+          loadedSegments+=("$element")
+        elif [[ -f "${p9kDirectory}/segments/custom/${element}.p9k" ]]; then # check if custom segment file exists
+          source "${p9kDirectory}/segments/custom/${element}.p9k" 2>&1
+          loadedSegments+=("$element")
+        fi
+      fi
+      "prompt_custom" "left" "$index" $joined $element[8,-1]
+    else
+      if [[ -z "${loadedSegments[(r)$element]}" ]]; then
```

Comment:
We talked about that already in https://github.com/onaforeignshore/powerlevel9k/commit/bbf597e5658eb9431e33bb2bd4ecbffed76c7941 and I still think this code is redundant. You now check for loaded segments on startup _and_ every time `buildLeftPrompt` is called.

Reply:
I haven't removed it yet because I have been waiting for your updates :)

Reply:
*NOTE* If we remove this, the tests fail because each test will need to be updated to load the correct segment definitions.

Reply:
Hmm. My Updates... Maybe we should put that question in a separate Issue, cause I think we should discuss it with more people.

Btw. good note.

Reply:
Okay. I opened #817 to discuss this. The suggestions there should be implemented in separate PRs, after this one was merged, IMHO.